### ISSUE TITLE
[Java] A FN about AvoidCalendarDateCreation 

### ISSUE THREAD

**Issue by Austin2088** [pmd/pmd#3402]
**Affects PMD Version:6.35.0** 

**Rule:[Avoid Calendar Date Creation](https://pmd.github.io/pmd-6.36.0/pmd_rules_java_performance.html#avoidcalendardatecreation)**

**Description:**
I think PMD needs to report a warning at line 6 because it calls Calendar.getTime() method, the program may, this can slow down the performance.

**Code Sample demonstrating the issue:**

```java
public class Example {
  public Calendar calendar;
  public void testAvoidCalendarDateCreation() {
    if(this.calendar == null)
       this.calendar = Calendar.getInstance();
    Date date = calendar.getTime(); // should report a warning here
  }
}
```

**Expected outcome:**

PMD should report a violation at line 6, but doesn't. This is a false-negative.

**Running PMD through:** *[Maven]*

**Comment by Zustin** [comment]
@adangel  Hi Andreas, thanks for your kind reply. However, when continue to test this case, I fina a problem, please refer to the following code.

PMD reports a waring at line 3, but not report warning at line 4. Actually method `getInstance` is static method and var calendar is final. The result should be consistent here and I think this is a false negative. Thanks for your kind help and consideration. I am willing to help you fix this issue.

```java
public final Calendar calendar = Calendar.getInstance();
public void foo() {
    Date date = calendar.getInstance().getTime(); // report a warning here
    // Date date = Calendar.getInstance().getTime(); // not report warning
}
```
